src/crt/sllshru.src

```diff
@@ -0,0 +1,51 @@
+; Performs unsigned 48 bit ("short long long") bit shift right
+; 
+; Arguments:
+; ude = Most significant bytes of arg 1
+; uhl = Least significant bytes of arg 1
+; ubc = Amount to shift by (effectively just shift by c)
+
+
+; Returns:
+; ude:uhl = ude:uhl>>c
+
+	assume	adl=1
+
+	section	.text
+	public	__sllshru
+__sllshru:
+	push bc
+	ld b, c
+	push de
+	push hl
+	ld hl, 0
+	add hl, sp
+	
+	inc hl
+	inc hl
+	inc hl
+	inc hl
+	inc hl
+	
+	srl (hl)
```

Comment:
This is up to you, but if you want to save a bit of space you could jump straight into the middle of the signed loop from here.